lib/spack/spack/environment/environment.py

```diff
@@ -2473,27 +2474,21 @@ def _equiv_dict(first, second):
     return same_values and same_keys_with_same_overrides
 
 
-def display_specs(concretized_specs):
-    """Displays the list of specs returned by `Environment.concretize()`.
+def display_specs(specs):
+    """Displays a list of specs traversed breadth-first, covering nodes, with install status.
 
     Args:
-        concretized_specs (list): list of specs returned by
-            `Environment.concretize()`
+        specs (list): list of specs
     """
-
-    def _tree_to_display(spec):
-        return spec.tree(
-            recurse_dependencies=True,
-            format=spack.spec.DISPLAY_FORMAT,
-            status_fn=spack.spec.Spec.install_status,
-            hashlen=7,
-            hashes=True,
-        )
-
-    for user_spec, concrete_spec in concretized_specs:
-        tty.msg("Concretized {0}".format(user_spec))
-        sys.stdout.write(_tree_to_display(concrete_spec))
-        print("")
+    tree_string = spack.spec.tree(
+        specs,
+        format=spack.spec.DISPLAY_FORMAT,
+        hashes=True,
+        hashlen=7,
+        status_fn=spack.spec.Spec.install_status,
+        key=traverse.by_dag_hash,
+    )
```

Comment:
(minor) Would the other calls to `spec.tree` not want to use these same arguments (e.g. `spack solve`), and likewise would this invocation not want to set `color=...`?

Also, given that this is not just a call to `tree`, it seems generic enough that it shouldn't live in `environment.py`.

Reply:
This one’s just the default output from `concretize()` and `install()`, so the options are meant to mirror what we’ve typically shown there. I think it’s here so that those two default concretization output cases are consistent.

`spec` and `solve` both have a lot of display options so they pass args to the `tree()` call. They’ve got the defaults that `spack spec` typically uses, so they’re consistent with each other.

I actually think all oftheee should probably have the same defaults — hashes and install status — but I’d rather consolidate that in a separate PR